src/astro/sql/operators/dataframe.py

```diff
@@ -91,70 +89,41 @@ def __init__(
             self.output_table: Optional[Table] = self.op_kwargs.pop("output_table")
         else:
             self.output_table = None
-        self.op_args = self.kwargs.get("op_args")  # type: ignore
+        self.op_args = self.kwargs.get("op_args", ())  # type: ignore
         self.identifiers_as_lower = identifiers_as_lower
 
         super().__init__(
             **kwargs,
         )
 
-    def handle_op_args(self):
-        full_spec = inspect.getfullargspec(self.python_callable)
-        op_args = list(self.op_args)
-        ret_args = []
-        for arg in op_args:
-            current_arg = full_spec.args.pop(0)
-            if (
-                full_spec.annotations[current_arg] == pd.DataFrame
-                and type(arg) == Table
-            ):
-                ret_args.append(
-                    _get_dataframe(arg, identifiers_as_lower=self.identifiers_as_lower)
-                )
-            else:
-                ret_args.append(arg)
-        self.op_args = tuple(ret_args)
-
-    def handle_op_kwargs(self):
-        param_types = inspect.signature(self.python_callable).parameters
-        self.op_kwargs = {
-            k: _get_dataframe(v, identifiers_as_lower=self.identifiers_as_lower)
-            if param_types.get(k).annotation == pd.DataFrame and type(v) == Table
-            else v
-            for k, v in self.op_kwargs.items()
-        }
-
     def execute(self, context: Dict):  # skipcq: PYL-W0613
-        self._set_variables_from_first_table()
-        self.handle_op_args()
-        self.handle_op_kwargs()
+        first_table = find_first_table(
+            op_args=self.op_args,  # type: ignore
+            op_kwargs=self.op_kwargs,
+            python_callable=self.python_callable,
+            parameters=self.parameters or {},  # type: ignore
+        )
+        if first_table:
+            self.conn_id = self.conn_id or first_table.conn_id  # type: ignore
+            self.database = self.database or first_table.metadata.database  # type: ignore
+            self.schema = self.schema or first_table.metadata.schema  # type: ignore
+        self.op_args = load_op_arg_table_into_dataframe(
+            self.op_args, self.python_callable, self.identifiers_as_lower
+        )
+        self.op_kwargs = load_op_kwarg_table_into_dataframe(
+            self.op_kwargs, self.python_callable, self.identifiers_as_lower
+        )
 
         pandas_dataframe = self.python_callable(*self.op_args, **self.op_kwargs)
         if self.output_table:
-            self.populate_output_table()
-            self.output_table.metadata.schema = (  # type: ignore
-                self.output_table.metadata.schema or SCHEMA
+            self.output_table.conn_id = self.output_table.conn_id or self.conn_id
+            db = create_database(self.output_table.conn_id or self.conn_id)
```

Comment:
```suggestion
            db = create_database(self.output_table.conn_id)
```

since we already handle the "or" condition on L119